src/host/HostMonitoringTemplate.cc

```diff
@@ -73,6 +74,173 @@ int SystemMonitoring::from_template(const Template &tmpl)
+        NUMAMonitoringNode node;
+        node.from_xml_node(node_xml);
+
+        unsigned int id;
+        node.get("NODE_ID", id);
+        nodes[id] = node;
+    }
+
+    xml.free_nodes(content);
+
+    return 0;
+}
+
+/* -------------------------------------------------------------------------- */
+/* -------------------------------------------------------------------------- */
+
+int NUMAMonitoring::from_template(const Template &tmpl)
+{
+    // Parse HugePages
+    vector<const VectorAttribute*> huge_pages;
+
+    tmpl.get("HUGEPAGE", huge_pages);
+
+    for (const auto* page : huge_pages)
+    {
+        int node_id;
+        unsigned long size;
+        unsigned long fr;
+
+        ostringstream oss;
+
+        if (page->vector_value("NODE_ID", node_id) != 0)
+        {
+            page->to_token(oss);
+
+            NebulaLog::warn("HMM", "Hugepage doesn't contain node ID: "
+                + oss.str());
+
+            continue;
+        }
+
+        if (page->vector_value("SIZE", size) != 0)
+        {
+            page->to_token(oss);
+
+            NebulaLog::warn("HMM", "Hugepage doesn't contain size: "
+                + oss.str());
+
+            continue;
+        }
+
+        page->vector_value("FREE", fr);
+
+        set_huge_page(node_id, size, fr);
+    }
+
+    // Parse Memory nodes
+    vector<const VectorAttribute*> mem_nodes;
+
+    tmpl.get("MEMORY_NODE", mem_nodes);
+
+    for (const auto* mem : mem_nodes)
+    {
+        int node_id;
+        unsigned long used;
+        unsigned long fr;
+
+        ostringstream oss;
+
+        if (mem->vector_value("NODE_ID", node_id) != 0)
+        {
+            mem->to_token(oss);
+
+            NebulaLog::warn("HMM", "Memory node doesn't contain node ID: "
+                + oss.str());
+
+            continue;
+        }
+
+        mem->vector_value("USED", used);
+        mem->vector_value("FREE", fr);
+
+        set_memory(node_id, used, fr);
+    }
+
+    return 0;
+}
+
+/* -------------------------------------------------------------------------- */
+/* -------------------------------------------------------------------------- */
+
+void NUMAMonitoring::set_huge_page(unsigned int node_id, unsigned long size, unsigned long fr)
+{
+    NUMAMonitoringNode* node;
+
+    auto it = nodes.find(node_id);
+    if ( it != nodes.end())
+    {
+        node = &it->second;
+    }
+    else
+    {
+        auto res = nodes.insert(make_pair(node_id, NUMAMonitoringNode()));
+        node = &res.first->second;
+        node->add("NODE_ID", node_id);
+    }
+
+    auto vatt = new VectorAttribute("HUGEPAGE");
+    vatt->replace("SIZE", size);
+    vatt->replace("FREE", fr);
+
+    node->set(vatt);
+}
+
+/* -------------------------------------------------------------------------- */
+/* -------------------------------------------------------------------------- */
+
+void NUMAMonitoring::set_memory(unsigned int node_id, unsigned long used, unsigned long fr)
+{
+    NUMAMonitoringNode* node;
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

We observed that the presence of non-empty statements like `a + b` without clear context or purpose within the codebase can lead to potential confusion or misinterpretation. To mitigate this issue, it&#39;s advisable to ensure that such statements are appropriately integrated into larger constructs that contribute meaningfully to the program&#39;s logic or functionality. This could involve assigning the result of the expression to a variable.